core/src/main/scala/org/apache/spark/Accumulable.scala

```diff
@@ -146,6 +198,32 @@ class Accumulable[R, T] private (
   def merge(term: R) { value_ = param.addInPlace(value_, term)}
 
   /**
+   * Merge in pending updates for ac consistent accumulators or merge accumulated values for
+   * regular accumulators.
+   */
+  private[spark] def internalMerge(term: Any) {
+    if (!consistent) {
+      merge(term.asInstanceOf[R])
+    } else {
+      mergePending(term.asInstanceOf[mutable.HashMap[(Int, Int), R]])
+    }
+  }
+
+  /**
+   * Merge another pending updates, checks to make sure that each pending update has not
+   * already been processed before updating.
+   */
+  private[spark] def mergePending(term: mutable.HashMap[(Int, Int), R]) = {
+    term.foreach{case ((rddId, splitId), v) =>
+      val splits = processed.getOrElseUpdate(rddId, new mutable.BitSet())
+      if (!splits.contains(splitId)) {
```

Comment:
add a doc saying that this isn't thread safe, but its ok b/c its only called w/in the dag scheduler event loop